docs/proposals/web-terminal.md

```diff
@@ -0,0 +1,23 @@
+**User Cluster WEB terminal**
+
+**Author**: Lukasz Zajaczkowski  (zreigz@gmail.com)
+
+**Status**: Draft proposal. Prototype in progress
+
+## Goals
+
+The KKP user have a WEB terminal interface where can access own clusters via CLI directly inside the browser. The Visual
+Web Terminal, located in the cluster details page. It allows running a commands that return Kubernetes resources. The 
+terminal has also other tools by default like helm, git, curl, wget.
+
+## Implementation
+
+Every user cluster has deployed Pod with the all required command line applications like: kubectl, helm, curl. From the backend
+the user can exec to this Pod and execute commands. The WEB terminal Pod has configured regular default user without root privileges.
+It's defined in Security Context for the Pod. Permission to access an object, like a file or command line applications, is based on user ID (UID) and group ID (GID).
+
+## Requirements
+
+ - For security reason the Pod with command line tools is deployed on the user cluster
+ - The kubeconfig needs to run as the login dex user (we don't want to expose admin kubeconfig).
```

Comment:
We should introduce a feature gate for this and allow admins to configure it via the UI as well.
<img width="888" alt="Screenshot 2022-06-07 at 11 20 16 AM" src="https://user-images.githubusercontent.com/18264334/172309633-6c88b371-a362-4b08-a6c9-eeed5e562722.png">